charts/corda/values.schema.json

```
@@ -0,0 +1,2703 @@
+                    ],
+                    "properties": {
+                        "host": {
+                            "format": "hostname",
+                            "type": "string",
+                            "default": "",
+                            "title": "the cluster database host",
+                            "examples": [
+                                ""
+                            ]
+                        },
+                        "additionalProperties": false,
+                        "type": {
+                            "type": "string",
+                            "default": "postgresql",
+                            "title": "the cluster database type",
+                            "examples": [
+                                "postgresql"
+                            ]
+                        },
+                        "port": {
+                            "type": "integer",
+                            "default": 5432,
+                            "title": "the cluster database port",
+                            "examples": [
+                                5432
+                            ]
+                        },
+                        "user": {
+                            "type": "string",
+                            "default": "user",
+                            "title": "the cluster database user",
+                            "examples": [
+                                "user"
+                            ],
+                            "minLength": 1
+                        },
+                        "database": {
+                            "type": "string",
+                            "default": "cordacluster",
+                            "title": "the name of the cluster database",
+                            "examples": [
+                                "cordacluster"
+                            ]
+                        },
+                        "password": {
+                            "type": "string",
+                            "default": "",
+                            "title": "the cluster database password (ignored if existingSecret is set, otherwise required)",
+                            "examples": [
+                                ""
+                            ]
+                        },
+                        "existingSecret": {
+                            "type": "string",
+                            "default": "",
+                            "title": "the name of an existing secret containing the cluster database password with a key of 'password'",
+                            "examples": [
+                                ""
+                            ]
+                        }
+                    },
+                    "oneOf": [
+                        {
+                            "required": ["password","existingSecret"
+                        ]
+                        }
+                    ],
```

`oneOf` takes a list of which at least one element must be satisfied. You only have one element in your list and it says that both values are required. I think you want something more like:
```suggestion
                    "oneOf": [
                        {
                            "required": ["password"]
                        },
                        {
                            "required": ["existingSecret"]
                        }
                    ],
```

---

If I do it this way, it doesn't reply with a significant error indicating which one to specify. For example, this is the error doing it the way above...
```bash
db.cluster: Must validate one and only one schema (oneOf)
```
Where as the error produced my way is 

```bash 
[debug] execution error at (corda/templates/cluster-db-secret.yaml:13:15): Must specify db.cluster.existingSecret or db.cluster.password
```

---

That second error isn't coming from the schema validation though - that's coming from the `required` function in `db-worker-secret.yaml`. That said, it's disappointing that the schema validation error is not more specific. We could just leave the requirement for those fields out of the schema entirely and rely on the runtime validation.

---

okay